File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.cs
public class Solution {
    public int AreaOfMaxDiagonal(int[][] dimensions) {
        double temp = 0;
        int ans = 0;
        foreach(var d in dimensions)
        {
            int x=d[0];
            int y=d[1];
            double tmp = Math.Sqrt(x*x+y*y);
            int an = x*y;
            if(tmp>temp)
            {
                temp=tmp;
                ans=an;
            }
            else if(tmp==temp)
            {
                ans=Math.Max(an, ans);
            }
        }
        return ans;
    }
}

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.py
class Solution(object):
    def areaOfMaxDiagonal(self, dimensions):
        """
        :type dimensions: List[List[int]]
        :rtype: int
        """
        temp=0
        ans=0
        for x,y in dimensions:
            l=x*x+y*y
            a=x*y
            if l>temp:
                temp=l
                ans=a
            elif l==temp:
                ans=max(ans,a)
        return ans